flycheck-haskell.el

```diff
@@ -164,6 +167,33 @@ database was not found."
     (when (re-search-forward flycheck-haskell-package-db-re nil 'noerror)
       (match-string 1))))
 
+(defconst flycheck-haskell-compiler-re
+  (rx line-start (zero-or-more (any space)) "with-compiler:"
+      (zero-or-more (any space))
+      (group (one-or-more (not (any space))))
+      (zero-or-more (any space) line-end))
+  "Regular expression to parse the compiler path.")
+
+(defun flycheck-haskell-get-compiler (cabal-config)
+  "Get the compiler path from CABAL-CONFIG.
+
+Return the compiler path as string, or nil, if the database was
+not found."
+  (with-temp-buffer
+    (insert-file-contents cabal-config)
+    (goto-char (point-min))
```

Comment:
I'm not sure that's correct... Reading through the `haskell-cabal` source, it looks like `haskell-cabal-get-setting` is meant to handle `*.cabal` files, but project-specific `cabal.config` files are in the format of `~/.cabal/config`.
Also, I can't seem to get it to work (although it does when I call it interactively).